docs/proposals/systemd-nodespec.md

```diff
@@ -0,0 +1,134 @@
+be started; under what circumstances it should be restarted; and any resource controls that should be applied to the service.
+* ```slice``` - a file that describes the inner-leaves of the cgroup hierarchy.  the name aligns with its placement in the hierarchy tree;
+for example, ```kubelet-besteffort.slice``` denotes a cgroup node ```/sys/fs/cgroup/<controller>/kubelet.slice/besteffort.slice```.  A
+service must attach to a single slice to denote where in the cgroup hierarchy the service is placed.
+* ```scope``` - a inner-leaf in the cgroup hierarchy that are created automatically by ```systemd``` to manage processes that are created
+externally.  For example, processes that are created by the ```docker``` daemon are placed in a ```scope``` in a particular ```slice```.
+
+By default, ```systemd``` defines three slices:
+
+* ```system.slice``` - system and application daemons
+* ```user.slice``` - user login related processes
+* ```machine.slice``` - virtual machines or containers that have init systems (not application space containers).
+
+For Kubernetes systems, we propose the kubelet should monitor the following by default:
+
+* ```system.slice``` - for cpu and memory usage by host daemons
+* ```machine.slice``` - for baremetal scenarios that may have super privileged containers or other vms on the node machine.
+
+It is assumed that ```kubelet``` and ```docker``` are launched via units in the ```system.slice```.
+
+The ```kubelet``` should detect the unit that has launched its service, and should discover the unit that manages the ```docker``` daemon.
+
+## Unified cgroup hierarchy
+
+Classical cgroup hierarchies were orthogonal per resource group controller.  For example, you could have one hierarchy for ```/sys/fs/cgroup/cpu/...``` and a completely different hierarchy for ```/sys/fs/cgroup/cpuacct``` even though one depended on the other.
+
+The direction is to move toward a single unified cgroup hierarchy, where the per-controller hierarchies are eliminated in favor of 
+just ```/sys/fs/cgroup/```.
+
+In this environment, there is automatic propagation of controllers, processes may only exist on leaf nodes in the hierarchy.
+
+```systemd``` has laid the ground work for enforcing a single unified control group hierarchy.
+
+It implicitly enforces a unified hierarchy by virtue of its representation of ```slice```, ```scope```, and ```service``` since a service
+only belongs to a single parent branch in the ```slice``` hierarchy.
+
+To align with the unified hierarchy, this document proposes the following set of ```slice``` definitions:
+
+* ```kubelet-besteffort.slice``` - cgroup that holds user containers with best effort quality of service
+* ```kubelet-burstable.slice``` - cgroup that holds user containers that have burstable quality of service
+* ```kubelet-burstable-guaranteed.slice```- cgroup that holds user containers that have guaranteed quality of service
```

Comment:
Would the 3 slices be statically defined, rather than configurable? How would the user select the qos tier? Can you give an example of what would be the cgroup configuration of each tier (mem&cpu)?